### ISSUE TITLE
Options.Default should be ignored for Flag

### ISSUE THREAD

**Issue by jasonwee** [akamensky/argparse#98]
```
package main

import (
	"fmt"
	"github.com/akamensky/argparse"
	"os"
)

func main() {
   parser := argparse.NewParser("accounts", "Prints provided string to stdout")
   noTest := parser.Flag("n", "no-test", &argparse.Options{Help: "boolean true/false", Default: true})

   err := parser.Parse(os.Args)
   if err != nil {
      fmt.Print(parser.Usage(err))
   }
   fmt.Println(*noTest)
}
```

why is this always true? the second should be false?
```
$ go run main.go 
true
$ go run main.go -n
true
$
```

**Comment by akamensky** [comment]
Flag default 'true' doesn't make sense. 

If flag is not provided it is false (default boolean value), if provided it is true. 

Setting default to true makes it to be always true.

I agree it can be improved by simply ignoring Options.Default for Flag.

**Comment by jasonwee** [comment]
Thanks for considering

**Comment by akamensky** [comment]
Resolved in #103

**Comment by jasonwee** [comment]
will upgrade and test this feature, thank you!
